staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -255,17 +61,17 @@ func WithRequestManagementByClient(
 		}
 
 		for {
-			rmState := reqMgmt.curState.Load().(*RMState)
-			fs := reqMgmt.pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
-			ps := reqMgmt.requestPriorityState(r, fs, rmState.priorityLevelStates)
-			if ps.config.Exempt {
+			rmState := reqMgmt.GetCurrentState()
+			fs := pickFlowSchema(r, rmState.GetFlowSchemas(), rmState.GetPriorityLevelStates())
+			ps := requestPriorityState(r, fs, rmState.GetPriorityLevelStates())
```

Comment:
Does it really make sense to have the functions `pickFlowSchema`, `requestPrioriryState`, `computeFlowDistinguisher`, and `hashFlowID` in this package after creating the `utilflowcontrol.Interface` abstraction?  What I currently see is that all the data has moved but not all the associated functionality.

Making those package functions rather than `reqmgmt` methods puts them in the `filters` package, so we have to think about whether those names make sense in that larger context.

Reply:
i moved these functions to the `utilflowcontrol` packages. the point of separating them from `utilflowcontrol.Interface` abstraction is to differ stateful functions from stateless functions. 

> Making those package functions rather than reqmgmt methods puts them in the filters package, so we have to think about whether those names make sense in that larger context.

yes, at least we should strip `http.Request` from those functions, the new package doesn't directly deal with the raw request.

Reply:
additionally,  `ComputeFlowDistinguisher` and `HashFlowID` can be squashed into one?

Reply:
So `utilflowcontrol.Interface` is really `utilflowcontrol.ConfigurationSource`, right?

`PickFlowSchema` clearly needs access to the HTTP request.  I made it available to `requestPriorityState` for the sake of defaulting logic.

Yes, `ComputeFlowDistinguisher` could be combined with `HashFlowID`.

Reply:
>  do you mean the digestion of the config API objects into config data structure?

yes, holding the instance of config data

> utilflowcontrol.Interface is really utilflowcontrol.ConfigurationSource, right?

probably, but it doesn't sound like a better name i guess, not only it is the config source but also it manages instances of fair-queuing systems? 

> PickFlowSchema clearly needs access to the HTTP request. I made it available to requestPriorityState for the sake of defaulting logic.

i refactored a bit by introducing a type named `RequestDigest`, which is a composition of user-identity and request-resources. how do you think about that?

> Yes, ComputeFlowDistinguisher could be combined with HashFlowID.

i nested hashing into `ComputeFlowDistinguisher`

Reply:
My problem was that `Interface` was not really the interface to all the functionality of that package, and so should have a more specific name.  In my latest review I suggested some mods that move the functionality not formerly covered by `Interface` into methods of objects reached from `Interface`, thereby making it a reasonable name.